Quantum-Control-Applications/Superconducting/Single Fixed Transmon/Use Case 3 - DRAG calibration/fine_DRAG.py

```diff
@@ -0,0 +1,154 @@
+from qm.qua import *
+from configuration import *
+from qm.QuantumMachinesManager import QuantumMachinesManager
+import matplotlib.pyplot as plt
+import numpy as np
+from scipy import signal
+
+n_avg = 2500 * 4
+
+cooldown_time = 5 * qubit_T1 // 4
+
+a_min = -1
+a_max = 1
+da = 0.01
+amps = np.arange(a_min, a_max + da / 2, da)  # + da/2 to add a_max to amplitudes
+
+with program() as drag2:
+    n = declare(int)
+    n_st = declare_stream()
+    a = declare(fixed)
+    I = declare(fixed)
+    Q = declare(fixed)
+    I_st = declare_stream()
+    Q_st = declare_stream()
+    state = declare(bool)
+    state_st = declare_stream()
+    state2_st = declare_stream()
+    I_g = declare(fixed)
+
+    with for_(n, 0, n < n_avg, n + 1):
+        # Notice it's + da/2 to include a_max (This is only for fixed!)
+        with for_(a, a_min, a < a_max + da / 2, a + da):
+            measure("readout", "resonator", None, dual_demod.full("rotated_cos", "out1", "rotated_sin", "out2", I_g))
+            with while_(I_g > -0.0003):
+                measure(
+                    "readout", "resonator", None, dual_demod.full("rotated_cos", "out1", "rotated_sin", "out2", I_g)
+                )
+            align()
+            wait(500)
+            # play('y90'*amp(a,0,0,1), 'qubit')
+            play("x180" * amp(1, 0, 0, a), "qubit")
+            play("y90" * amp(a, 0, 0, 1), "qubit")
+            align("qubit", "resonator")
+            measure(
+                "readout",
+                "resonator",
+                None,
+                dual_demod.full("rotated_cos", "out1", "rotated_sin", "out2", I),
+                dual_demod.full("rotated_minus_sin", "out1", "rotated_cos", "out2", Q),
+            )
+            save(I, I_st)
+            save(Q, Q_st)
+            assign(state, I > ge_threshold)
+            save(state, state_st)
+            # wait(cooldown_time, "resonator")
+
+            align()
+
+            measure("readout", "resonator", None, dual_demod.full("rotated_cos", "out1", "rotated_sin", "out2", I_g))
+            with while_(I_g > -0.0003):
+                measure(
+                    "readout", "resonator", None, dual_demod.full("rotated_cos", "out1", "rotated_sin", "out2", I_g)
+                )
+            align()
+            wait(1000)
```

Comment:
Why do you need this? And why is it different from the one above?

Reply:
my bad, should have been the same both

Reply:
Ok makes sense